src/uucore/src/lib/features/checksum.rs

```diff
@@ -308,7 +311,21 @@ fn determine_regex(lines: &[String]) -> Option<(Regex, bool)> {
     ];
 
     for line in lines {
-        let line_trim = line.trim();
+        let mut line_trim = os_str_as_bytes(line).expect("UTF-8 decoding failed");
+        // FIXME: Replace this with `.trim_ascii()` when MSRV is 1.80
+        while let Some(first) = line_trim.first() {
+            if !first.is_ascii_whitespace() {
+                break;
+            }
+            line_trim = &line_trim[1..];
+        }
+        while let Some(last) = line_trim.last() {
+            if !last.is_ascii_whitespace() {
+                break;
+            }
+            line_trim = &line_trim[..line_trim.len() - 1];
```

Comment:
could you please document what this line is doing, it isn't obvious

Reply:
Will do.

It is a simple implementation of the `trim_ascii()` function which is not available for MSRV 1.70.